### PR TITLE
Bug 1957295: All platform pods should have system-* or openshift-user-critical priority class

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
In order to avoid conditions like https://github.com/openshift/cluster-network-operator/pull/994

cc @smarterclayton @soltysh @damemi @ingvagabund

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
cc @wking

**Comment by smarterclayton** [comment]
/hold

we actually want to move these out of system-* in most cases.  Also, we need to update the guidance in enhancements CONVENTIONS.md doc as well.  Let’s sync tomorrow (but this will absolutely be needed).

File: test/extended/pods/priorityclasses.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	It("ensure platform components have system-* priority class associated", func() {
+		// iterate over the references to find valid images
+		pods, err := oc.KubeFramework().ClientSet.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Failf("unable to list pods: %v", err)
+		}
+
+		// list of pods that use images not in the release payload
+		invalidPodPriority := sets.NewString()
+		// a pod in a namespace that begins with kube-* or openshift-* must come from our release payload
```
Comment (aravindhp) [inline test/extended/pods/priorityclasses.go:27]: This needn't be true for `openshift-*` namespaces as some OLM managed operators user this also.

Comment (ravisantoshgudimetla) [reply test/extended/pods/priorityclasses.go:27]: If you're just referring to the comment, I don't mind removing it. However please note that this test suite when run in origin wouldn't include those OLM managed components unless explicitly deployed.

Comment (aravindhp) [reply test/extended/pods/priorityclasses.go:27]: Yes I am just referring to the comment. You can reword it to clarify given you are suggesting in your email that OLM managed operators can also pull in this test.

**Comment by markmc** [comment]
> Also, we need to update the guidance in enhancements CONVENTIONS.md doc as well.

Yes. Apparently the guidance now is:

```
* If it is fine for your operator/operand to be preempted by user workload specify and OOMKilled  use openshift-user-critical priority class
* If you want your operator/operand not to be preempted by user-workload but still be OOMKilled use system-cluster-critical priority class
* If you want operator/operand not be preempted by user-workload and OOMKilled last use system-node-critical priority class
```

**Review by soltysh — APPROVED** [review]
Consistent with https://github.com/openshift/enhancements/pull/763
/hold cancel
/lgtm
/approve
